finetuner/data.py

```diff
@@ -160,22 +163,38 @@ def load_finetune_data_from_csv(
                     options.create_point_clouds,
                     point_cloud_size=options.point_cloud_size,
                 )
-
-            if d2 is None:
-                d1.tags[DEFAULT_TAG_KEY] = label
-                yield d1
-            elif (d1.text and d2.text) or (d1.uri and d2.uri):
-                # same modality
-                d1.tags[DEFAULT_TAG_KEY] = artificial_label
-                d2.tags[DEFAULT_TAG_KEY] = artificial_label
-                artificial_label += 1
-                yield d1
-                yield d2
-            else:
-                # different modalities, for CLIP
-                d1.modality = t1
-                d2.modality = t2
-                yield Document(chunks=[d1, d2])
+                if col1 in queries and col2 in queries:
+                    continue
+                if col1 in queries:
+                    queries[col2] = queries[col1]
+                    d2.tags[DEFAULT_TAG_KEY] = queries[col1]
+                    # only yield d2
+                elif col2 in queries:
+                    queries[col1] = queries[col2]
+                    d1.tags[DEFAULT_TAG_KEY] = queries[col1]
```

Comment:
I don't really understand this condition. col2 should not contain queries right?

Reply:
To me there are several cases where queries and documents are interchangeable. In text-text search does it really matter which column contains the query and which one contains the document? The resulting DocumentArray will be the same either way